src/main/java/org/gwtbootstrap3/extras/toggleswitch/client/ui/base/ToggleSwitchBase.java

```diff
@@ -206,21 +209,59 @@ public void setValue(final Boolean value) {
         setValue(value, false);
     }
 
+    /**
+     * Called when this widget is attached to the DOM.
+     * 
+     * Here we want to make sure the state of the toggle switch hasn't changed before it was 
+     * attached or re-attached to the DOM.  If the value has changed update the toggle switch 
+     * via {@link #switchState(Element, boolean, boolean)}.
+     */
+    @Override
+    public void onAttach() {
+        super.onAttach();
+        boolean switchValue = switchState(getElement());
+        boolean value = getValue();
+        if (value != switchValue) {
+            switchState(getElement(), value, true);
+        }
+    }
+    
```

Comment:
Why do not call ONLY once switchState() ? It ensures the value synchronization between toggleSwitch and input. If they are different, you don't need to call it twice.
````
// This method is a workaround of the native JS issue : #...
// It should be removed when it gets fixed...
@Override
public void onAttach() {
  super.onAttach();
  // Ensure synchronization between DOM value and toggle switch value
  switchState(getElement(), getValue(), true);
}
```